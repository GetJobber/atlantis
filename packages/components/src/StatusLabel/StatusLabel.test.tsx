import React from "react";
import { cleanup, render } from "@testing-library/react";
import { StatusLabel } from ".";

afterEach(cleanup);

it("renders an inactive StatusLabel", () => {
  const { container } = render(<StatusLabel label="Foo" status="inactive" />);
  expect(container).toMatchSnapshot();
});

it("renders a successful StatusLabel", () => {
  const { container } = render(<StatusLabel label="Foo" status="success" />);
  expect(container).toMatchSnapshot();
});

it("renders a warning StatusLabel", () => {
  const { container } = render(<StatusLabel label="Foo" status="warning" />);
  expect(container).toMatchSnapshot();
});

it("renders a critical StatusLabel", () => {
  const { container } = render(<StatusLabel label="Foo" status="critical" />);
  expect(container).toMatchSnapshot();
});

it("renders an informative StatusLabel", () => {
  const { container } = render(
    <StatusLabel label="Foo" status="informative" />,
  );
  expect(container).toMatchSnapshot();
});

it("renders an end-aligned StatusLabel", () => {
  const { container } = render(
    <StatusLabel label="Foo" status="informative" alignment="end" />,
  );
  expect(container).toMatchSnapshot();
});
