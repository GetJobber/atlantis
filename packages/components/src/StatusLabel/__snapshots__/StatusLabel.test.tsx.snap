// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a critical StatusLabel 1`] = `
<div>
  <div
    class="statusLabelRow critical"
    role="status"
  >
    <div
      class="statusIndicatorWrapper"
    >
      <span
        class="statusIndicator"
        data-testid="ATL-Status-Indicator-critical"
      />
    </div>
    <p
      class="base regular small text"
    >
      Foo
    </p>
  </div>
</div>
`;

exports[`renders a successful StatusLabel 1`] = `
<div>
  <div
    class="statusLabelRow success"
    role="status"
  >
    <div
      class="statusIndicatorWrapper"
    >
      <span
        class="statusIndicator"
        data-testid="ATL-Status-Indicator-success"
      />
    </div>
    <p
      class="base regular small text"
    >
      Foo
    </p>
  </div>
</div>
`;

exports[`renders a warning StatusLabel 1`] = `
<div>
  <div
    class="statusLabelRow warning"
    role="status"
  >
    <div
      class="statusIndicatorWrapper"
    >
      <span
        class="statusIndicator"
        data-testid="ATL-Status-Indicator-warning"
      />
    </div>
    <p
      class="base regular small text"
    >
      Foo
    </p>
  </div>
</div>
`;

exports[`renders an end-aligned StatusLabel 1`] = `
<div>
  <div
    class="statusLabelRow informative labelTextEndAligned"
    role="status"
  >
    <div
      class="statusIndicatorWrapper"
    >
      <span
        class="statusIndicator"
        data-testid="ATL-Status-Indicator-informative"
      />
    </div>
    <p
      class="base regular small text end"
    >
      Foo
    </p>
  </div>
</div>
`;

exports[`renders an inactive StatusLabel 1`] = `
<div>
  <div
    class="statusLabelRow inactive"
    role="status"
  >
    <div
      class="statusIndicatorWrapper"
    >
      <span
        class="statusIndicator"
        data-testid="ATL-Status-Indicator-inactive"
      />
    </div>
    <p
      class="base regular small text"
    >
      Foo
    </p>
  </div>
</div>
`;

exports[`renders an informative StatusLabel 1`] = `
<div>
  <div
    class="statusLabelRow informative"
    role="status"
  >
    <div
      class="statusIndicatorWrapper"
    >
      <span
        class="statusIndicator"
        data-testid="ATL-Status-Indicator-informative"
      />
    </div>
    <p
      class="base regular small text"
    >
      Foo
    </p>
  </div>
</div>
`;
